src/pegen/tokenizer.py

```diff
@@ -60,6 +60,17 @@ def diagnose(self) -> tokenize.TokenInfo:
             self.getnext()
         return self._tokens[-1]
 
+    def get_last_non_whitespace_token(self) -> tokenize.TokenInfo:
+        index = self._index - 1
+        while index >= 0:
+            tok = self._tokens[index]
+            if tok.type != tokenize.ENDMARKER and (
+                tok.type < tokenize.NEWLINE or tok.type > tokenize.DEDENT
+            ):
+                break
+            index -= 1
+        return tok
```

Comment:
You can probably simplify this using `itertools.islice` and a reversed iterator as you are basically iterating backwards and checking a condition on the given element.

Reply:
I pushed a change using reversed but I am not sure why I would need islice. 

Also could you please wait on this one for me to add some tests for the tokenizer specifically ?

Reply:
> why I would need islice.

In theory to filter anything on the right of `self._index - 1`

> Also could you please wait on this one for me to add some tests for the tokenizer specifically ?

Sure

Reply:
Just slicing the list of _tokens directly as I did in my last commit should work just fine, no ? I may not have time to work on tests tonight (I am in France) but I will do my best to address those during the week.